sdks/go/node/datadir/datadir.go

```diff
@@ -21,10 +22,10 @@ type DataDir interface {
 func ensureExists(
 	resolvedDataDirPath string,
 ) error {
-	if err := os.MkdirAll(resolvedDataDirPath, 0775|os.ModeSetgid); err != nil {
-		return err
-	}
-	return nil
+	// don't de-privilege group
+	unix.Umask(0002)
```

Comment:
OOC - would the following have the same effect? Or would it prevent the Docker VM (or some other user/group) from being able to read the file?
```suggestion
	unix.Umask(0000)
```

Reply:
this would work as well it just is relaxed beyond what we need; it would also remove the mask from "other"  and we only need it removed for "group"

Reply:
Ahh I see - I thought the `0` mask in the 'other' position would remove any 'other' perms but it sounds a `0` just allows any existing perms to persist. Thanks for clarifying!